### PR TITLE
Add OperatorFilterer into ERC721MOnft

### PR DESCRIPTION
**Pull request by channing-magiceden**
- Remove functions not being used in `IERC721M`
- Further downsize `IERC721MLite` by removing crossminting and updating specific stages
- Add OperatorFilterer into `IERC721MLite` 
- Bump package version from `0.0.10` to `0.0.11`

image

### CODE REVIEW

**Review by mi-yu — APPROVED** [review]

File: contracts/ERC721M.sol [hunk 1]
```diff
@@ -110,13 +102,6 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
         _;
     }
 
-    /**
-     * @dev Returns cosigner address.
-     */
-    function getCosigner() external view override returns (address) {
```
Comment (Henry-MagicEden) [inline contracts/ERC721M.sol:116]: Do these need to be removed from the main `ERC721M` contract? Could we just remove them from IERC721M but keep them in the actual contract?

Comment (channing-magiceden) [reply contracts/ERC721M.sol:116]: It's not quite useful and ME has no references. We don't have any use case that could change cosigner. And we don't have issues historically to mis-set cosigner. Maybe good to keep them for debugging convenience but I think it's easier to just reset `cosigner`.

File: contracts/ERC721M.sol [hunk 2]
```diff
@@ -94,14 +94,6 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
         _;
     }
 
-    /**
-     * @dev Returns whether NOT mintable.
-     */
-    modifier cannotMint() {
```
Comment (nothing0012) [inline contracts/ERC721M.sol:100]: why remove this? not used anymore?

File: contracts/ERC721M.sol [hunk 3]
```diff
@@ -129,14 +114,7 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
...
         _cosigner = cosigner;
-        emit SetCosigner(cosigner);
-    }
-
-    /**
-     * @dev Returns expiry in seconds.
-     */
-    function getTimestampExpirySeconds() public view override returns (uint64) {
-        return _timestampExpirySeconds;
+        emit SetCosigner(_cosigner);
```
Comment (nothing0012) [inline contracts/ERC721M.sol:117]: let's use cosigner to avoid a storage read

File: contracts/ERC721M.sol [hunk 4]
```diff
@@ -535,18 +505,6 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
         emit PermanentBaseURI(_currentBaseURI);
     }
 
-    /**
-     * @dev Returns token URI suffix.
-     */
-    function getTokenURISuffix()
```
Comment (nothing0012) [inline contracts/ERC721M.sol:541]: we need this, right?

File: contracts/ERC721M.sol [hunk 5]
```diff
@@ -147,19 +125,12 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
...
-        return _crossmintAddress;
-    }
-
     /**
      * @dev Sets crossmint address if using crossmint. This allows the specified address to call `crossmint`.
      */
     function setCrossmintAddress(address crossmintAddress) external onlyOwner {
         _crossmintAddress = crossmintAddress;
-        emit SetCrossmintAddress(crossmintAddress);
+        emit SetCrossmintAddress(_crossmintAddress);
```
Comment (nothing0012) [inline contracts/ERC721M.sol:133]: this uses more gas, is it really going to save the binary size?

File: contracts/ERC721M.sol (hunk 2 above)
Comment (qianmao) [reply contracts/ERC721M.sol:100]: Yeah not used.

Comment (channing-magiceden) [reply contracts/ERC721M.sol:100]: Yeah not used.

File: contracts/ERC721M.sol (hunk 4 above)
Comment (channing-magiceden) [reply contracts/ERC721M.sol:541]: Not quite, we never use this externally.

**Review by nothing0012 — APPROVED** [review]

**Pull request merged**